sbnana/CAFAna/Core/SpectrumLoader.cxx

```diff
@@ -599,6 +1003,63 @@ namespace ana
       } // end for slice
     } // end for spillcut
 
+    // Universe knobs, Truth
+    for(caf::SRTrueInteractionProxy& nu: sr->mc.nu){
+      // Some shifts only adjust the weight, so they're effectively nominal,
+      // but aren't grouped with the other nominal histograms. Keep track of
+      // the results for nominals in these caches to speed those systs up.
+      CutVarCache<bool, TruthCut, caf::SRTrueInteractionProxy> nomTruthCutCache;
+
+      for ( auto& [shift, truthcutmap] : fTruthNUniversesTreeDefs ) {
+        // Need to provide a clean slate for each new set of systematic
+        // shifts to work from. Copying the whole StandardRecord is pretty
+        // expensive, so modify it in place and revert it afterwards.
+        caf::SRProxySystController::BeginTransaction();
+
+        bool shifted = false;
+
+        double systWeight = 1;
+        // Can special-case nominal to not pay cost of Shift()
+        if(!shift.IsNominal()){
+          shift.Shift(&nu, systWeight);
+          // If there were only weighting systs applied then the cached
+          // nominal values are still valid.
+          shifted = caf::SRProxySystController::AnyShifted();
+        }
+
+        for ( auto& [truthcut, treemap] : truthcutmap ) {
+          const bool pass = shifted ? truthcut(&nu) : nomTruthCutCache.Get(truthcut, &nu);
+          // Cut failed, skip all the histograms that depended on it
+          if(!pass) continue;
```

Comment:
Unlike NSigmas tree, we here ask if the nu passes _after_ the SystShift. (Note that the same thing is done in the slice version so it's not just here.)

I'm thinking about if this is actually the way we want it.

I guess if only weight-based systematics are used in the NUniverses Tree we should see the same behaviour.

Reply:
See comment below, this is good to me.